libr/core/cmd_type.c

```diff
@@ -259,26 +259,57 @@ static void save_parsed_type(RCore *core, const char *parsed) {
 	}
 }
 
-//TODO
-//look at the next couple of functions
-//can be optimized into one right ... you see it you do it :P
-static int stdprintifstruct(void *p, const char *k, const char *v) {
-	if (!strncmp (v, "struct", strlen ("struct") + 1)) {
-		r_cons_println (k);
+// TODO these are going to be moved in sdb
+static int __cmp_asc(const void *a, const void *b) {
+	const SdbKv *ka = a;
+	const SdbKv *kb = b;
+	return strcmp (sdbkv_key (ka), sdbkv_key (kb));
+}
+
+struct foreach_list_filter_t {
+	SdbForeachCallback filter;
+	SdbList *list;
+};
+
+static int foreach_list_filter_cb(void *user, const char *k, const char *v) {
+	struct foreach_list_filter_t *u = (struct foreach_list_filter_t *)user;
+	SdbList *list = u->list;
+	if (!u->filter || u->filter (NULL, k, v)) {
+		SdbKv *kv = R_NEW0 (SdbKv);
+		kv->key = strdup (k);
```

Comment:
check for null

Reply:
I'd prefer if you can review these functions in the related sdb PR.